detectors/node/opentelemetry-resource-detector-instana/src/detectors/InstanaAgentDetector.ts

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Detector,
+  Resource,
+  ResourceDetectionConfig,
+} from '@opentelemetry/resources';
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
+import * as http from 'http';
+
+class InstanaAgentDetector implements Detector {
+  readonly INSTANA_AGENT_DEFAULT_HOST = 'localhost';
+  readonly INSTANA_AGENT_DEFAULT_PORT = 42699;
+
+  async detect(_config?: ResourceDetectionConfig): Promise<Resource> {
+    const host =
+      process.env.INSTANA_AGENT_HOST || this.INSTANA_AGENT_DEFAULT_HOST;
+    const port = Number(
+      process.env.INSTANA_AGENT_PORT || this.INSTANA_AGENT_DEFAULT_PORT
+    );
+
+    const data = await this._retryHandler(host, port, 0);
+
+    return new Resource({
+      [SemanticResourceAttributes.PROCESS_PID]: data.pid,
+      'instana.host.id': data.agentUuid,
+      // NOTE: The agent reads this property to correlate the process with the service
+      'service.instance.id': data.pid.toString(),
+    });
+  }
+
+  private timeout(ms: number) {
+    return new Promise(resolve => setTimeout(resolve, ms));
```

Comment:
Can this timeout block the process from proper shutdown?
In other places we usually [`unref`](https://nodejs.org/api/timers.html#timeoutunref) so opentelemetry will  not hold the event loop active